lib/elixir/lib/enum.ex

```diff
@@ -803,11 +803,13 @@ defmodule Enum do
   @spec filter_map(t, (element -> as_boolean(term)),
     (element -> element)) :: list
 
-  def filter_map(enumerable, filter, mapper) when is_list(enumerable) do
+  def filter_map(enumerable, filter, mapper)
+  when is_list(enumerable) and is_function(filter, 1) and is_function(mapper, 1) do
```

Comment:
@lexmag I have to deal with `with` all the time, and it is very common in `with` to have several lines bellow. So what I do, with add two spaces after `with` to make number of spaces even.